project-koku/koku
# Daily archives minor bug fixes + code cleanup

## Jira Ticket

[COST-####](https://issues.redhat.com/browse/COST-####)

## Description

Changes/improvements:

1.  Add one final check for multiple workers when saving the start date to the manifest
2. Move `clear_s3_files` outside the if csv files statement so we clear parquet files even if we have no csv files (small edge case)
3. Remove the now unnecessary manifest stored file name batch numbers from the ocp on cloud parquet files (file names are unique so no longer required)
4. Remaining changes are cleaning up and re-factoring the code to make it more accurate (naming, function args etc)

## Testing

1. Checkout Branch
5. Restart Koku
6. Run Make commands to create + load test customer data
7. Everything should ingest as expected.

## Notes
Advanced testing per improvement above:
1. Add a sleep statement before we call `set_manifest_daily_start_date` long enough that you can manually set the entry in the db. Then see its not overridden by this function and instead returns the manually set value. `update reporting_common_costusagereportmanifest set daily_archive_start_date = date('2023-09-01') where provider_id = '56456ceb-0da2-4cfa-9d48-8264ae52a2c1';`
3. Create OCP on AWS sources/data, Delete all AWS csv files from minio, Create new AWS data, hit download, See all parquet files are removed.
4. Create OCP on AWS sources/data, See that all ocp on cloud parquet files are created with the expected file names. (can run before and after this change to confirm files are the same)
5. No additional testing
...
